ip_country/src/country_finder.rs

```diff
@@ -0,0 +1,301 @@
+                0x605C0100AB76020E,
+                0x0000000000000000,
+            ],
+            271,
+        )
+    }
+
+    pub(crate) fn ipv6_country_data() -> (Vec<u64>, usize) {
+        (
+            vec![
+                0x3000040000400007,
+                0x00C0001400020000,
+                0xA80954B000000700,
+                0x4000000F0255604A,
+                0x0300004000040004,
+                0xE04AAC8380003800,
+                0x00018000A4000001,
+                0x2AB0003485C0001C,
+                0x0600089000000781,
+                0xC001D20700007000,
+                0x00424000001E04AA,
+                0x15485C0001C00018,
+                0xC90000007812AB00,
+                0x2388000700006002,
+                0x000001E04AAC00C5,
+                0xC0001C0001801924,
+                0x0007812AB0063485,
+                0x0070000600C89000,
+                0x1E04AAC049D23880,
+                0xC000180942400000,
+                0x12AB025549BA0001,
+                0x0040002580000078,
+                0xAC8B800038000300,
+                0x000000000001E04A,
+            ],
+            1513,
+        )
+    }
+
+    #[test]
+    fn finds_ipv4_address_in_fourth_block() {
+        let result = CountryCodeFinder::find_country(
+            &COUNTRY_CODE_FINDER_TEST,
+            IpAddr::from_str("1.0.6.15").unwrap(),
+        );
+        //crate::country_finder::COUNTRY_CODE_FINDER.ipv4.remove(COUNTRY_CODE_FINDER.ipv4.len() - 1);
+        assert_eq!(result, Some(Country::try_from("AU").unwrap()))
+    }
+
+    #[test]
+    fn does_not_find_ipv4_address_in_zz_block() {
+        let timestart = SystemTime::now();
+        let result = CountryCodeFinder::find_country(
+            &COUNTRY_CODE_FINDER_TEST,
+            IpAddr::from_str("0.0.5.0").unwrap(),
+        );
+        let timeend = SystemTime::now();
+
+        assert_eq!(result, None);
+        let duration = timeend.duration_since(timestart).unwrap().as_secs();
+        println!("duration: {}", duration);
+        assert!(duration <= 5);
+    }
+
+    #[test]
+    fn finds_ipv6_address_in_fourth_block() {
+        let result = CountryCodeFinder::find_country(
+            &COUNTRY_CODE_FINDER_TEST,
+            IpAddr::from_str("1:0:5:0:0:0:0:0").unwrap(),
+        );
+
+        assert_eq!(result, Some(Country::try_from("AU").unwrap()))
+    }
+
+    #[test]
+    fn does_not_find_ipv6_address_in_zz_block() {
+        let result = CountryCodeFinder::find_country(
+            &COUNTRY_CODE_FINDER_TEST,
+            IpAddr::from_str("0:0:5:0:0:0:0:0").unwrap(),
+        );
+
+        assert_eq!(result, None)
+    }
+
+    #[test]
+    fn real_test_ipv4_with_google() {
+        let result = CountryCodeFinder::find_country(
+            &COUNTRY_CODE_FINDER,
+            IpAddr::from_str("142.250.191.132").unwrap(), // dig www.google.com A
+        )
+        .unwrap();
+
+        assert_eq!(result.name, "United States".to_string());
```

Comment:
It seems like the lower situated tests use more assertions than this one. They also assert on the free world, and ISO code.